### PR TITLE
Automation for new TC-CGEN-2.4

### PR DESCRIPTION
**Pull request by cecille**
Note that this includes a separate change to fix revoke commissioning function. Putting up so I can link in the test plans.

### CODE REVIEW

File: src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp [hunk 1]
```diff
@@ -193,6 +193,8 @@ bool emberAfAdministratorCommissioningClusterRevokeCommissioningCallback(
 {
     ChipLogProgress(Zcl, "Received command to close commissioning window");
 
+    Server::GetInstance().GetFailSafeContext().ForceFailSafeTimerExpiry();
```
Comment (bzbarsky-apple) [inline src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:196]: The spec doesn't say to do this, right?  Do we need a spec fix here?  Seems like we do.

Comment (cecille) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:196]: I think the spec could use some more concrete wording here. Right now, it's quite dependent on your interpretation of what it means to "revoke" a commissioning window and "expire" the pake verifier. The sdk takes the pake verifier expiration to include all open pase sessions, but it then leaves the failsafe in a bad state where there's a failsafe open that's not attached to any ongoing session.

I think the spec could use some more detail here, but I'm not sure that expiring the pase sessions is a bug. It does seem like a bug that the failsafe doesn't get closed off when the pase session expires and perhaps the spec needs to be more concrete here as well - if a session gets closed, perhaps the spec should be explicit that the associated failsafe gets expired.

Comment (bzbarsky-apple) [reply src/app/clusters/administrator-commissioning-server/administrator-commissioning-server.cpp:196]: Thank you for filing https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6215

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/python/ChipDeviceController-ScriptBinding.cpp [hunk 2]
```diff
@@ -538,8 +547,12 @@ PyChipError pychip_DeviceController_OpenCommissioningWindow(chip::Controller::De
     if (option == Controller::CommissioningWindowOpener::CommissioningWindowOption::kTokenWithRandomPIN)
     {
         SetupPayload payload;
-        return ToPyChipError(Controller::AutoCommissioningWindowOpener::OpenCommissioningWindow(
-            devCtrl, nodeid, System::Clock::Seconds16(timeout), iteration, discriminator, NullOptional, NullOptional, payload));
+        auto opener =
```
Comment (tehampson) [inline src/controller/python/ChipDeviceController-ScriptBinding.cpp:550]: What is responsible for freeing this? I am OK with adding a TODO comment later on that does this

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**